src/functionalTest/java/uk/gov/hmcts/reform/userprofileapi/DeleteRolesToExistingUserFuncTest.java

```diff
@@ -0,0 +1,113 @@
+        RestAssured.useRelaxedHTTPSValidation();
+        idamClient = new IdamClient(configProperties);
+    }
+
+    @Test
+    public void should_delete_user_profile_with_roles_successfully() throws Exception {
+
+
+        CreateUserProfileData data = createUserProfileData();
+        List<String> roles = new ArrayList<>();
+        roles.add(puiUserManager);
+        String email = idamClient.createUser(roles);
+
+        data.setEmail(email);
+        createUserProfile(data, HttpStatus.CREATED);
+
+        RoleName role1 = new RoleName(puiOrgManager);
+        Set<RoleName> rolesName = new HashSet<>();
+        rolesName.add(role1);
+        UpdateUserProfileData userProfileData = new UpdateUserProfileData();
+        userProfileData.setRolesAdd(rolesName);
+
+        GetUserProfileResponse resource =
+                testRequestHandler.sendGet(
+                        requestUri + "?email=" + email.toLowerCase(),
+                        GetUserProfileResponse.class
+                );
+
+        LOG.info("before addroles call");
+        UserProfileRolesResponse resource1 =
+                testRequestHandler.sendPut(
+                        userProfileData,
+                            HttpStatus.OK,
+                           requestUri + "/" + resource.getIdamId(), UserProfileRolesResponse.class);
+
+        LOG.info("after addroles call" + resource1);
+
+        GetUserProfileWithRolesResponse resource2 =
+                testRequestHandler.sendGet(
+                        "/v1/userprofile/" + resource.getIdamId() + "/roles",
+                        GetUserProfileWithRolesResponse.class
+                );
+        LOG.info("Roles addroles call" + resource2);
+        assertThat(resource2.getRoles().size()).isNotNull();
+        assertThat(resource2.getRoles().size()).isEqualTo(3);
+        RoleName roleDelete = new RoleName(puiOrgManager);
+        Set<RoleName> rolesDelete = new HashSet<>();
+        rolesDelete.add(roleDelete);
+
+        UpdateUserProfileData userProfileDataDelete = new UpdateUserProfileData();
+        userProfileDataDelete.setRolesDelete(rolesDelete);
+
+        UserProfileRolesResponse deleteResourceResp =
+                testRequestHandler.sendDelete(
+                        userProfileDataDelete,
+                        HttpStatus.OK,
+                        requestUri + "/" + resource.getIdamId(), UserProfileRolesResponse.class);
+
+        LOG.info("after DeleteRole call" + deleteResourceResp);
+
+
+        GetUserProfileWithRolesResponse resourceForDeleteCheck =
+                testRequestHandler.sendGet(
+                        "/v1/userprofile/" + resource.getIdamId() + "/roles",
+                        GetUserProfileWithRolesResponse.class
+                );
+        LOG.info("Roles addroles call" + resource2);
+        assertThat(resourceForDeleteCheck.getRoles().size()).isNotNull();
+        assertThat(resourceForDeleteCheck.getRoles().size()).isEqualTo(2);

```

Comment:
Better to check if exact roles are deleted from SIDAM, just check in GetUserProfileWithRolesResponse if roles deleted are vanished from SIDAM and the roles which were not deleted are still intact in SIDAM.

Reply:
Yes. Included for roles check as well.